kolibri/core/assets/src/core-store.js

```diff
@@ -94,6 +98,15 @@ const mutations = {
     state.core.logging.session.total_time_at_last_save = timeSpent;
     state.core.logging.session.progress_at_last_save = progress;
   },
+  SET_LOGGING_MASTERY_STATE(state, masteryState) {
+    state.core.logging.mastery = masteryState;
+  },
+  SET_LOGGING_ATTEMPT_STATE(state, attemptState) {
+    state.core.logging.attempt = attemptState;
+  },
+  SET_EMPTY_LOGGING_STATE(state) {
+    state.core.logging = baseLoggingState;
```

Comment:
Curious if this works as expected. In the context of vuex & reactivity, seems potentially problematic to re-use the same object rather than making a copy? I'm not sure though.